src/data-manager/mongo/mongo-data-manager.ts

```
@@ -220,12 +220,17 @@ const updateItemInCollection = async (
       .collection(collectionName)
       .updateOne({ _id: objectId }, { $set: updateObject });
 
-    if (matchedCount === 1 && modifiedCount === 1) {
+    if (matchedCount === 1) {
+      if (modifiedCount == 0) {
+        Log.warn(`No changes made for item with id ${id} in ${collectionName}`);
+      }
```

it seems that if matchedCount === 1 and modifiedCount === 0  we log the warning but still perform the operation.  is this the intended idea?

---

The intended idea is not to throw errors when the document already has the identical values as the updates, which will result in modifiedCount == 0; 

As we discussed in our meeting, an alternative could be added in the future to update the "updatedAt" properties of a document so that the modifiedCount will still be 1.

---

Yes I rememeber that.  I am not saying we should throw an error merely wondering if we still need to perfom the call to the db to update

---

I see. The update should have already been attempted by this point. If we want to prevent the update db update operation, we will first have to check whether the values are the same already, either in the mongo manager or the database manager. Is that what you have in mind?